module/hgd.sh

```diff
@@ -321,13 +351,11 @@ function :hgd:save() {
         local session="$2"
         local hgd="$3"
 
-        local -a hosts
-        hosts=( $(:hgd:resolve ${tldid} ${hgd}) )
-        if [ $? -eq 0 -a ${#hosts[@]} -gt 0 ]; then
-            :hgd:delete ${session}
-        fi
+        :hgd:delete ${session}
+
+        local -a hosts=( $(:hgd:resolve ${tldid} ${hgd}) )
         echo -ne "${session}\t${tldid}\t${NOW?}\t${hgd}\t${hosts[@]}\n" >> ${g_HGD_CACHE?}
-        e=${CODE_SUCCESS?}
+        [ ${#hosts[@]} -eq 0 ] || e=${CODE_SUCCESS?}
```

Comment:
In Uni they told us to make it:
[ ${#hosts[@]} -gt 0 ] && e=${CODE_SUCCESS?}
But since its not really an if statement its probably fine as well :-) ... Just wanted to say something